x-pack/plugins/alerting/server/saved_objects/migrations/7.16/index.ts

```diff
@@ -34,15 +34,15 @@ function getRemovePreconfiguredConnectorsFromReferencesFn(
 }
 
 function getCorrespondingAction(
-  actions: SavedObjectAttribute,
+  actions: RawRuleAction | RawRuleAction[],
   connectorRef: string
 ): RawRuleAction | null {
   if (!Array.isArray(actions)) {
     return null;
   } else {
     return actions.find(
-      (action) => (action as RawRuleAction)?.actionRef === connectorRef
-    ) as RawRuleAction;
+      (action) => (action as unknown as RawRuleAction)?.actionRef === connectorRef
```

Comment:
The cast is not needed anymore. It can be `actions.find((action) => action?.actionRef === connectorRef) ?? null;`.